### PR TITLE
Merge of MDL dev branch.

### PR DESCRIPTION
**Pull request by niklasharrysson**
Update #840

### CODE REVIEW

File: libraries/stdlib/stdlib_defs.mtlx [hunk 1]
```diff
@@ -2575,7 +2575,7 @@
   <nodedef name="ND_rotate3d_vector3" node="rotate3d" nodegroup="math">
     <input name="in" type="vector3" value="0.0, 0.0, 0.0"/>
     <input name="amount" type="float" value="0.0" unittype="angle" unit="degree"/>
-    <parameter name="axis" type="vector3" value="0.0, 1.0, 0.0"/>
+    <input name="axis" type="vector3" value="0.0, 1.0, 0.0"/>
```
Comment (bernardkwok) [inline libraries/stdlib/stdlib_defs.mtlx:2578]: We'll need upgrade code for this.

File: source/MaterialXTest/CMakeLists.txt [hunk 2]
```diff
@@ -36,6 +36,18 @@ file(GLOB materialx_headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 
 list(APPEND materialx_headers "${CMAKE_CURRENT_SOURCE_DIR}/Catch/catch.hpp")
 
+if (NOT MATERIALX_BUILD_GEN_ARNOLD)
```
Comment (bernardkwok) [inline source/MaterialXTest/CMakeLists.txt:39]: This should not be restored. There is a scan for each directory now to find files to include (add_tests above).

File: source/MaterialXView/Editor.cpp [hunk 3]
```diff
@@ -458,6 +458,7 @@ void PropertyEditor::addItemToForm(const mx::UIPropertyItem& item, const std::st
             }
         });
         v1->setSpinnable(editable);
+        v1->setEditable(editable);
```
Comment (bernardkwok) [inline source/MaterialXView/Editor.cpp:461]: This looks like a merge mistake?

File: libraries/stdlib/stdlib_defs.mtlx (hunk 1 above)
Comment (niklasharrysson) [reply libraries/stdlib/stdlib_defs.mtlx:2578]: Good point! Will fix.

File: source/MaterialXView/Editor.cpp (hunk 3 above)
Comment (niklasharrysson) [reply source/MaterialXView/Editor.cpp:461]: I don't think so, it's for v1 here and v2 below.

Setting it to editable makes it possible to edit the value by typing in a number, in addition to spinning the up and down arrow on the control.

**Pull request merged**